sulfnucrate.F90

```diff
@@ -4,51 +4,386 @@
 
 !!  Calculates particle production rates due to nucleation <rhompe>:
 !!  binary homogeneous nucleation of sulfuric acid and water only
-!!  Numerical method follows Zhao & Turco, JAS, V.26, No.5, 1995.
+!!  Numerical method follows one of the following:
+!!   1. Zhao & Turco, JAS, V.26, No.5, 1995.
+!!   2. Vehkamaki, H., M. Kulmala, I. Napari, K.E.J. Lehtinen,
+!!       C. Timmreck, M. Noppel and A. Laaksonen, 2002,
+!!       An improved parameterization for sulfuric acid-water nucleation
+!!       rates for tropospheric and stratospheric conditions,
+!!       J. Geophys. Res., 107, 4622, doi:10.1029/2002jd002184
 !!
-!!  This was moved from sulfnuc to make the code more manageable.
 !!
 !!  @author Mike Mills, Chuck Bardeen
-!!  @version Jun-2013
-subroutine sulfnucrate(carma,cstate, iz, igroup, h2o, h2so4, beta1, beta2, ftry, rstar, nucbin, nucrate, rc) 
+!!  @version May-2022
+subroutine sulfnucrate(carma, cstate, iz, igroup, h2o, h2so4, beta1, beta2, ftry, radius_cluster, nucbin, nucrate, rc)
   use carma_precision_mod
   use carma_enums_mod
   use carma_constants_mod
   use carma_types_mod
   use carmastate_mod
   use carma_mod
   use sulfate_utils
-  
+
   implicit none
-  
-  type(carma_type), intent(in)         :: carma       !! the carma object
-  type(carmastate_type), intent(inout) :: cstate      !! the carma state object
-  integer, intent(in)                  :: iz          !! level index
-  integer, intent(in)                  :: igroup      !! group index
-  real(kind=f), intent(out)            :: h2o         !! H2O concentrations in molec/cm3 
-  real(kind=f), intent(out)            :: h2so4       !! H2SO4 concentrations in molec/cm3
+
+  type(carma_type), intent(in)         :: carma          !! the carma object
+  type(carmastate_type), intent(inout) :: cstate         !! the carma state object
+  integer, intent(in)                  :: iz             !! level index
+  integer, intent(in)                  :: igroup         !! group index
+  real(kind=f), intent(out)            :: h2o            !! H2O concentrations in molec/cm3
+  real(kind=f), intent(out)            :: h2so4          !! H2SO4 concentrations in molec/cm3
   real(kind=f), intent(out)            :: beta1
   real(kind=f), intent(out)            :: beta2
   real(kind=f), intent(out)            :: ftry
-  real(kind=f), intent(out)            :: rstar       !! critical radius (cm)
-  integer, intent(out)                 :: nucbin      !! bin in which nucleation occurs
-  real(kind=f), intent(out)            :: nucrate     !! nucleation rate #/x/y/z/s
-  integer, intent(inout)               :: rc          !! return code, negative indicates failure
+  real(kind=f), intent(out)            :: radius_cluster !! critical radius (cm)
```

Comment:
Why was name change from rstar to radius cluster? It is rstar in the place that calls this - sulfnuc.F90. I am not a fan of changing the parameter names in the middle as it makes searches harder. I am okay if zhao or vehkamaki want to have different terminology in their routines, but would rather have it be te same before that.